gemrb/tests/2DAImporter/Test_2DAImporter.cpp

```diff
@@ -0,0 +1,119 @@
+/* GemRB - Infinity Engine Emulator
+ * Copyright (C) 2023 The GemRB Project
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; either version 2
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#include <gtest/gtest.h>
+
+#include "../../core/Streams/FileStream.h"
+#include "../../plugins/2DAImporter/2DAImporter.h"
+
+namespace GemRB {
+
+static const path_t SAMPLE_FILE = PathJoin("tests", "resources", "2DAImporter", "sample.2da");
+static const path_t ENC_SAMPLE_FILE = PathJoin("tests", "resources", "2DAImporter", "sample_encrypted.2da");
+
+class p2DAImporter_Test : public testing::TestWithParam<path_t> {
+protected:
+	FileStream* stream;
+	p2DAImporter unit;
+public:
+	void SetUp() override {
+		this->stream = new FileStream{};
+
+		stream->Open(GetParam());
+		assert(unit.Open(stream));
+		// p2DAImporter deletes stream internally after use
```

Comment:
I dislike this design. I vote for RAII or a `std::unqiue_ptr` signature to make this clear at code level.